lib/rml-generator.js

```diff
@@ -166,21 +170,158 @@ class RMLGenerator extends AbstractGenerator {
+        const formSubject = namedNode(this.baseIRI + this.getUniqueID('form'));
+        const propertyAffordanceSubject = namedNode(this.baseIRI + this.getUniqueID('propertyAffordance'));
+        const securitySubject = namedNode(this.baseIRI + this.getUniqueID('security'));
+
+        this.prefixes['td'] = namespaces.td
+        this.prefixes['wotsec'] = namespaces.wotsec
+        this.prefixes['hctl'] = namespaces.hctl
+        this.prefixes['idsa'] = namespaces.idsa
 
-      this.quads.push(quad(
-        sSubject,
-        namedNode(namespaces.rml + 'source'),
-        databaseSubject
-      ));
+        this.quads.push(quad(
+          sSubject,
+          namedNode(namespaces.rml + 'source'),
+          wotSubject
+        ));
 
-      this.quads.push(quad(
-        sSubject,
-        namedNode(namespaces.rml + 'query'),
-        literal(source.query)
-      ));
+        /* Build td:Thing */
+        this.quads.push(quad(
+          wotSubject,
+          namedNode(namespaces.rdf + 'type'),
+          namedNode(namespaces.td + 'Thing')
+        ));
+
+        /* Build td:PropertyAffordance */
+        this.quads.push(quad(
+          propertyAffordanceSubject,
+          namedNode(namespaces.rdf + 'type'),
+          namedNode(namespaces.td + 'PropertyAffordance')
+        ));
+
+        this.quads.push(quad(
+          propertyAffordanceSubject,
+          namedNode(namespaces.td + 'hasForm'),
+          formSubject
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.rdf + 'type'),
+          namedNode(namespaces.td + 'Form')
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'hasTarget'),
+          literal(source.access)
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'forContentType'),
+          literal(source.contentType)
+        ));
+
+        let operationType = null;
+        switch(source.operationType) {
+          case 'read':
+            operationType = namedNode(namespaces.td + 'readproperty');
+            break
+            
+          case 'write':
+          default:
+            break
+        }
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'hasOperationType'),
+          operationType
+        ));
+
+        this.quads.push(quad(
+          wotSubject,
+          namedNode(namespaces.td + 'hasPropertyAffordance'),
+          propertyAffordanceSubject
+        ));
 
-      this._generateDatabaseDescription(databaseSubject, source);
+        // FIXME: W3C WoT Protocol Bindings
+
+        /* Build wotsec:$SCHEMESecurityScheme */
+        if (source.security) {
+          if (Array.isArray(source.security)) {
+            Object.keys(source.security).forEach(index => {
+              let security = source.security[index]
+              // FIXME: support more security schemes
```

Comment:
Can you make a separate issue for this?

Reply:
Done: https://github.com/RMLio/yarrrml-parser/issues/119